internal/workerutil/worker.go

```diff
@@ -261,17 +263,21 @@ func (w *Worker) dequeueAndHandle() (dequeued bool, err error) {
 	log15.Debug("Dequeued record for processing", "name", w.options.Name, "id", record.RecordID())
 
 	if hook, ok := w.handler.(WithHooks); ok {
-		hook.PreHandle(handleCtx, record)
+		preCtx, endObservation := w.options.Metrics.operations.preHandle.With(workerCtxWithSpan, nil, observation.Args{})
```

Comment:
I think this should still use handleCtx, as the `cancel` function will be invoked for that on job cancel, not for `w.ctx`.
```suggestion
		preCtx, endObservation := w.options.Metrics.operations.preHandle.With(handleCtx, nil, observation.Args{})
```